src/App.tsx

```diff
@@ -1,54 +1,89 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+const debounce = (func:React.Dispatch<React.SetStateAction<string>>,
+  delay:number) => {
+  let timerId: number;
+
+  return (...args:string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(func, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selected, setSelected] = useState<Person>();
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const visiblePeople = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(query.toLowerCase()),
+    );
+  }, [peopleFromServer, appliedQuery]);
```

Comment:
The dependency array for useMemo should include 'query', not 'appliedQuery' or 'peopleFromServer'. 'peopleFromServer' is not a state or prop and 'appliedQuery' is not used in the memoization function. The function is using 'query' variable which is not included in the dependency array.